src/mixxxapplication.cpp

```diff
@@ -154,7 +155,8 @@ bool MixxxApplication::notify(QObject* target, QEvent* event) {
         break;
     }
     case QEvent::MouseButtonRelease: {
-        QMouseEventEditable* mouseEvent = static_cast<QMouseEventEditable*>(event);
+        QMouseEventEditable* mouseEvent =
+                static_cast<QMouseEventEditable*>(event); // clazy:exclude=wrong-qevent-cast
```

Comment:
I just looked at this recently, this actually triggers UBSAN, so we should find an alternative to `QMouseEventEditable` and this cast.

Reply:
I agree. Fortunately this file seems to be the only occurrence so we should enable that clazy rule to avoid adding it elsewhere.

Reply:
Qt seems to have a similar issue here: https://bugreports.qt.io/browse/QTBUG-99563

Reply:
Normally a downcast to a inherited object is undefined, because the object might be bigger and we might access memory out of the boundary of the original object. I think the only way around the undefined behavior is to make a copy, modify the value and than copy the result back. That is a CPU cost to fix an non-issue   ... :-/

Reply:
I don't think these are particularly expensive... I'd take the hit for avoiding UB. It would be very annoying to use ubsan with "false positives" like this.

Reply:
I opened an issue for this: https://github.com/mixxxdj/mixxx/issues/12013